dts/bindings/interrupt-controller/shared-irq.yaml

```diff
@@ -6,7 +6,24 @@ include: [interrupt-controller.yaml, base.yaml]
 
 properties:
   interrupts:
-      required: true
+    required: true
 
   label:
-      required: true
+    required: true
+
+  num-clients:
```

Comment:
We should figure this out by scanning the interrupt devicetree rather than having a property for it.

Reply:
Info can be obtained using `DT_SUPPORTS_DEP_ORDS`

Reply:
I have updated it now to use DT_INST_SUPPORTS_DEP_ORDS to get the number of clients.
Is there a more straight forward solution than to count them with a for_each?